### DISCUSSION TITLE
warning: "pgm_read_word" redefined (code from 02/04)

### DISCUSSION THREAD

**Discussion by daveclifford** [DCC-EX/EX-T3-WiFi#52] [General]
Executing task: platformio run --environment firebeetle2 
Processing firebeetle2 (platform: espressif32@^5.3.0; board: firebeetle2; framework: arduino)
--------------------------------------------------------------------------------------------------------------
In file included from .pio/libdeps/firebeetle2/QRcode_eSPI/src/qrcode_espi.cpp:2:

.pio/libdeps/firebeetle2/QRcodeDisplay/src/qrencode.h:5: warning: "pgm_read_word" redefined
 #define pgm_read_word(x) *x
 
In file included from /.platformio/packages/framework-arduinoespressif32/cores/esp32/WString.h:29,
                 from /.platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:166,
                 from .pio/libdeps/firebeetle2/QRcode_eSPI/src/qrcode_espi.cpp:1:
/.platformio/packages/framework-arduinoespressif32/cores/esp32/pgmspace.h:39: note: this is the location of the previous definition
 #define pgm_read_word(addr) ({ \
'
'
'
Successfully created esp32 image.

**Comment by alex-code** [comment]
The QR lib I used defines a macro with the same name.
The author of that should probably have an if defined check but this only produces a warning.

**Reply by daveclifford** [reply]
I learned something from you about compiling code with same names in VSCode and PlatformIO.

Maybe others don't care about "program warnings", but I worked within a large group (5-10) of programmers,\. Some were of the type - the compiler will find my errors - I was one who spent the time to "fix before found" and was known to create "tight" code - using register to register instructions (much faster than memory to register instructions. SSome of my code is still running after 50 years!!!!

Dave
